From 011f3e7fbc405a4c51550234add54ac903433b60 Mon Sep 17 00:00:00 2001
From: Ian Kilpatrick <ikilpatrick@chromium.org>
Date: Wed, 22 May 2024 16:29:37 +0000
Subject: [PATCH] [anchor] Fix anchor-center crash.

We had a NOTREACHED within grid for anchor-center when it was valid.
Ensure it maps to center.

Bug: 341928664
Change-Id: If9fa88f9febb31333d3d646dfce356cc1456259e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5552987
Commit-Queue: Mason Freed <masonf@chromium.org>
Reviewed-by: Mason Freed <masonf@chromium.org>
Auto-Submit: Ian Kilpatrick <ikilpatrick@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1304494}
---

--- a/third_party/blink/renderer/core/layout/grid/grid_item.cc
+++ b/third_party/blink/renderer/core/layout/grid/grid_item.cc
@@ -95,6 +95,7 @@
       }
       return is_for_columns ? logical.InlineEnd() : logical.BlockEnd();
     }
+    case ItemPosition::kAnchorCenter:
     case ItemPosition::kCenter:
       return AxisEdge::kCenter;
     case ItemPosition::kFlexStart:
@@ -126,9 +127,6 @@
     case ItemPosition::kAuto:
       NOTREACHED();
       return AxisEdge::kStart;
-    case ItemPosition::kAnchorCenter:
-      NOTREACHED();
-      return AxisEdge::kCenter;
   }
 }
 
--- /dev/null
+++ b/third_party/blink/web_tests/external/wpt/css/css-anchor-position/grid-anchor-center-crash.html
@@ -0,0 +1,5 @@
+<!DOCTYPE html>
+<link rel="help" href="https://issues.chromium.org/issues/341928664">
+<div style="display: grid; position: relative;">
+  <div style="position: absolute; align-self: anchor-center;"></div>
+</div>
